data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/ReactiveProducerFactory.java

```diff
@@ -34,4 +35,13 @@
      * @return a new ReactiveKafkaProducer
      */
     public ReactiveKafkaProducer<K, V> create(Vertx v, Properties config);
+
+    /**
+     * Create a new ReactiveKafkaProducer
+     *
+     * @param v        the Vertx instance used when creating the vertx KafkaProducer
+     * @param producer the Kafka producer
+     * @return a new ReactiveKafkaProducer
+     */
+    public ReactiveKafkaProducer<K, V> create(Vertx v, Producer<K, V> producer);
```

Comment:
Can you describe why do we need 2 methods instead of only 1?

Reply:
I think we might only need the one with the producer in input and we can remove the other one, and replace the usage in `ReceiverVerticleFactory` with something like this?

```java
properties -> kafkaProducerFactory.create(v, new KafkaProducer<>(properties))
```

Reply:
I have provided two methods to create producer
one for
https://github.com/knative-sandbox/eventing-kafka-broker/blob/898e479be20128819351ca341aff9d4e6f8ba50e/data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/main/ReceiverVerticleFactory.java#L66

and another one for 
https://github.com/knative-sandbox/eventing-kafka-broker/blob/898e479be20128819351ca341aff9d4e6f8ba50e/data-plane/receiver/src/test/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticleTracingTest.java#L111

Yes we can have ` kafkaProducerFactory.create(v, new KafkaProducer<>(properties))` and wrap it up in the vertex module using vertx `kafkaProducer.create(vertx v, Producer p)`

Reply:
done